nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -43,22 +43,26 @@ use uuid::Uuid;
 impl DataStore {
     /// Stores a new sled in the database.
     ///
+    /// Returns the sled, and whether or not it was updated on success.
+    ///
     /// Returns an error if `sled_agent_gen` is stale, or the sled is
     /// decommissioned.
     pub async fn sled_upsert(
         &self,
         sled_update: SledUpdate,
-    ) -> CreateResult<Sled> {
+    ) -> CreateResult<(Sled, bool)> {
         use db::schema::sled::dsl;
         // required for conditional upsert
         use diesel::query_dsl::methods::FilterDsl;
 
-        diesel::insert_into(dsl::sled)
-            .values(sled_update.clone().into_insertable())
+        let insertable_sled = sled_update.clone().into_insertable();
+        let now = insertable_sled.time_modified().clone();
+
+        let sled = diesel::insert_into(dsl::sled)
+            .values(insertable_sled)
             .on_conflict(dsl::id)
             .do_update()
             .set((
-                dsl::time_modified.eq(Utc::now()),
                 dsl::ip.eq(sled_update.ip),
```

Comment:
One other realization - are we actually updating `time_modified` in the on-conflict case? Looks like no?

Reply:
I realized I commented this while looking at the right hand side, but it is here on the left hand side.

Why are we no longer updating time modified here? Doesn't that mean that even if we do the update, `time_modified` won't change?

Reply:
Oh, whoops. I totally biffed that. I was thinking everything in the` insertable_sled` would go in. I'll add that back. I'm actually going to revert most of these changes, since as you point out below the SledIdentity already contains the modified value. That means none of the tests need to change.

Reply:
Oy, I'm tired. Tests still need to change. Anyway, have a fix. Thanks for pointing all this out.

Reply:
> Why are we no longer updating time modified here? Doesn't that mean that even if we do the update, time_modified won't change?

When I added the redundant `time_modified` field to sled, I assumed that would take precedence. It probably would have just been completely ignored. Thanks for pointing out `identity`. I reverted this behavior in [9c6efef](https://github.com/oxidecomputer/omicron/pull/5526/commits/9c6efef90c4c01fc3a74d27dbeb7ae8f13517c0b)
except the `now` value is reused so we don't have different value on the conflict case.